lib/ui/recommendation.js

```diff
@@ -44,12 +57,26 @@ RecommendationRow.prototype = {
       image.setAttribute('src', row.data.enhancements.domain.logo);
       return image;
     }
-    return null;
+    return row.renderLetterbox(row);
+  },
+
+  _firstLetterInDomain: function(url) {
+    const domain = url.replace(/^[^\/]+\/+/, '').split('/')[0].split('.');
+    const firstLevelDomain = domain[domain.length - 2];
+    return firstLevelDomain[0];
```

Comment:
Simpler, more reliable way to get the initial letter of the domain name:
```js
// first, get an nsIURI from the url
let uri = Services.ios.newURI(url, null, null);
// next, use the TLD service to get the base domain
let baseDomain = Services.eTLD.getBaseDomain(uri);
let initialLetter = baseDomain[0];
```
The advantage of using the existing services is that they account for all the quirks / edge cases in global domain names and TLDs.